### PR TITLE
refactor(CNV-47217): use golang gzip library

### PR DESCRIPTION
**Pull request by codingben**
Use golang's gzip library directly from
the source code, in order to be able to
test this functionality as well.

### CODE REVIEW

**Comment by codingben** [comment]
@ksimon1 Can you please review it?

File: main.go [hunk 1]
```diff
@@ -83,15 +85,30 @@ func downloadVirtualMachineDiskImage(vmName, volumeName string) error {
...
+	defer diskImg.Close()
 
-	if err := cmd.Run(); err != nil {
+	gzipReader, err := gzip.NewReader(diskImg)
+	if err != nil {
+		return err
+	}
+	defer gzipReader.Close()
+
+	newDiskImg, err := os.Create(diskPathDecompressed)
```
Comment (ksimon1) [inline main.go:100]: Does this new image has correct owner and permissions?

Comment (codingben) [reply main.go:100]: I've adjusted it to have `0666` that grants read and write permission to everyone.

**Comment by codingben** [comment]
@ksimon1 Please let me know if something is missing and I'll take care of it in a new PR. We'll remove this functionality later.

**Pull request merged**